services/models.py

```diff
@@ -183,22 +258,53 @@ def delete_gdpr_data(self, api_token: str, dry_run=False):
         profile on the related GDPR API.
 
         Dry run parameter can be used for asking the service if delete is possible.
-        An exception will be raised by this method if deletion response from the
-        service indicates an error or if GDPR related URLs have not been configured
-        for the related service.
+        Returns a dict with keys "service", "dry_run", "success", and "errors".
+        The errors content from the service is returned if the service provides a JSON
+        response with an "errors" key containing valid error content.
         """
+        result = {
+            "service": self.service,
+            "dry_run": dry_run,
+            "success": False,
+            "errors": [],
+        }
+
+        url = self.service.get_gdpr_url_for_profile(self.profile)
+
         data = {}
         if dry_run:
             data["dry_run"] = True
 
-        url = self.service.get_gdpr_url_for_profile(self.profile)
-        if url:
+        try:
             response = requests.delete(
                 url, auth=BearerAuth(api_token), timeout=5, data=data
             )
-            response.raise_for_status()
-            return True
+        except requests.RequestException:
+            return _add_error_to_result(
+                result,
+                SERVICE_GDPR_API_REQUEST_ERROR,
+                gettext_noop(
+                    "Error when making a request to the GDPR URL of the service"
+                ),
```

Comment:
As noted, this project doesn't have translations. In these error cases, where a "synthetic" error is produced, can we just leave the message out? Any client consuming the response may have to search for these special error codes anyway and act accordingly. That can include showing localized messages, but it would be the concern of the client.

Reply:
I'd keep the text. The client can of course then ignore it.

Reply:
We can introduce them later if needed? Added the message in english for now.

Reply:
Roger that